src/Tools/DynamoInstallDetective/Utilities.cs

```diff
@@ -88,19 +88,22 @@ public static IEnumerable FindProductInstallations(string productSearchPattern,
         /// info.</returns>
         public static IEnumerable FindMultipleProductInstallations(List<string> productSearchPatterns, string fileSearchPattern)
         {
-            var installs = new InstalledProducts();
-            // Look up products with ASM installed on user's computer
-            foreach (var productSearchPattern in productSearchPatterns)
+            using (RegUtils.StartCache())
```

Comment:
I wrapped existing code in a using (RegUtils.StartCache()) that will allow us to cache the Registry records within the scope